prepare/cards/rag/end_to_end/clapnq.py

```diff
@@ -0,0 +1,217 @@
+import itertools
+import math
+import random
+from dataclasses import dataclass
+
+import pandas as pd
+from unitxt import add_to_catalog
+from unitxt.blocks import TaskCard, TemplatesDict
+from unitxt.loaders import LoadCSV, LoadFromDictionary
+from unitxt.operators import Copy, ListFieldValues, Set
+from unitxt.templates import InputOutputTemplate
+from unitxt.test_utils.card import test_card
+
+
+@dataclass(frozen=True)
+class ClapNqBenchmarkConstants:
+    # Raw_data
+    TRAIN_RAW_FILE_URL: str = "https://raw.githubusercontent.com/primeqa/clapnq/main/retrieval/train/question_train_answerable.tsv"
+    TEST_RAW_FILE_URL: str = "https://raw.githubusercontent.com/primeqa/clapnq/main/retrieval/dev/question_dev_answerable.tsv"
+
+    # Fields
+    ID: str = "id"
+    QUESTION: str = "question"
+    DOC_ID_LIST: str = "doc-id-list"
+    ANSWERS: str = "answers"
+
+
+@dataclass(frozen=True)
+class ClapNqDocumentsConstants:
+    # Raw_data
+    RAW_FILE_URL: str = "https://media.githubusercontent.com/media/primeqa/clapnq/main/retrieval/passages.tsv"
+
+    # Fields
+    ID: str = "id"
+    TEXT: str = "text"
+    TITLE: str = "title"
+
+    ARTIFACT_NAME: str = "cards.rag.documents.clap_nq.en"
+
+
+@dataclass(frozen=True)
+class SubsetCreationConstants:
+    # For subset creation
+    SUBSET_SIZE = 100
+    RANDOM_SEED = 43
+    NEGATIVE_DOCUMENTS_FACTOR = 4
+
+
+# We want to create a subset of ClapNq - based on a subset of questions from the benchmark.
+random.seed(SubsetCreationConstants.RANDOM_SEED)
+# 1 - first we read the benchmark and the documents
+benchmark_train_df = pd.read_csv(ClapNqBenchmarkConstants.TRAIN_RAW_FILE_URL, sep="\t")
+benchmark_test_df = pd.read_csv(ClapNqBenchmarkConstants.TEST_RAW_FILE_URL, sep="\t")
+documents_df = pd.read_csv(ClapNqDocumentsConstants.RAW_FILE_URL, sep="\t")
+
+# 2 - We compute the size of train subset and test subset
+ratio_train_test = len(benchmark_train_df) / len(benchmark_test_df)
+train_size = math.floor(SubsetCreationConstants.SUBSET_SIZE / ratio_train_test)
+test_size = SubsetCreationConstants.SUBSET_SIZE - train_size
```

Comment:
not sure i follow here, 
suppose `len(benchmark_train_df)=100` and `len(benchmark_test_df)=25`,
then `ratio_train_test=4` and we get
```
train_size=25
test_size=75
```
(assuming i get it right)
it this ok?

Reply:
Good catch! fixing that! 

Anyway we will introduce that in a separate approach in a next PR .